hybrasyl/Monolith.cs

```diff
@@ -133,52 +136,171 @@ private static void SpawnMonster(Monster monster, Map map)
+                                    }
+                                }
                             }
                             else
                             {
-                                //movey
+                                //cast
+                                monster.Shout("I SHOULD BE CASTING RIGHT NOW!");
                             }
                         }
                         else
                         {
-                            //cast
+                            //check facing and attack or cast
+
+                            var nextAction = _random.Next(1, 6);
+                            if (nextAction > 1)
+                            {
+                                var facing = monster.CheckFacing(monster.Direction, closest);
+                                if (facing)
+                                {
+                                    monster.AssailAttack(monster.Direction, closest);
+                                }
+                            }
+                            else
+                            {
+                                monster.Shout("I WANT TO CAST BUT ITS NOT IMPLEMENTED YET!");
+                            }
                         }
                     }
-                    else
+                }
+            }
+            if (monster.ShouldWander)
+            {
+                var nextAction = _random.Next(0, 2);
+
+                if (nextAction == 1)
+                {
+                    var nextMove = _random.Next(0, 4);
+                    switch (nextMove)
+                    {
+                        case 0:
+                            monster.Walk(Direction.East);
+                            break;
+                        case 1:
+                            monster.Walk(Direction.West);
+                            break;
+                        case 2:
+                            monster.Walk(Direction.North);
+                            break;
+                        case 3:
+                            monster.Walk(Direction.South);
+                            break;
+                    }
+                }
+                else
+                {
+                    var nextMove = _random.Next(0, 4);
+                    switch (nextMove)
                     {
-                        //check facing and attack or cast
+                        case 0:
+                            monster.Turn(Direction.East);
+                            break;
+                        case 1:
+                            monster.Turn(Direction.West);
+                            break;
+                        case 2:
+                            monster.Turn(Direction.North);
+                            break;
+                        case 3:
+                            monster.Turn(Direction.South);
+                            break;
```

Comment:
I believe this could be simpler:

```
if (monster.ShouldWander)
{
    var nextAction = _random.Next(0, 2);
    var nextMove = (Direction)_random.Next(0, 4);

    if (nextAction == 1)
    {
        monster.Walk(nextMove);
    }
    else
    {
        monster.Turn(nextMove);
    }
}
```

Reply:
Could be for sure.